File: PizzaStore/src/PizzaStore.Domain/Entities/Order.cs
namespace PizzaStore.Domain.Entities;

public class Order : BaseEntity
{
    public string UserId { get; set; } = string.Empty;
    public ApplicationUser User { get; set; } = null!;
    
    public decimal TotalPrice { get; set; }
    public OrderStatus Status { get; set; }
    
    public DateTime? ConfirmedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? CancelledAt { get; set; }
    
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: PizzaStore/frontend/src/types/order.ts
export type OrderStatus =
  | 'Pending'
  | 'Confirmed'
  | 'Preparing'
  | 'OutForDelivery'
  | 'Delivered'
  | 'Cancelled'

export interface OrderItemTopping {
  id: string
  toppingId: string
  toppingNameAtOrder: string
  toppingPriceAtOrder: number
}

export interface OrderItem {
  id: string
  orderId: string
  pizzaVariantId: string
  pizzaNameAtOrder: string
  pizzaSizeAtOrder: string
  pizzaBasePriceAtOrder: number
  quantity: number
  specialInstructions: string | null
  subtotalAtOrder: number
  toppings: OrderItemTopping[]
}

export interface Order {
  id: string
  userId: string
  totalPrice: number
  status: OrderStatus
  createdAt: string
  confirmedAt: string | null
  completedAt: string | null
  cancelledAt: string | null
  items: OrderItem[]
}
